frontend/packages/dev-console/src/actions/modify-helm-release.ts

```diff
@@ -0,0 +1,20 @@
+import { coFetchJSON } from '@console/internal/co-fetch';
+import { history } from '@console/internal/components/utils';
+import { deleteResourceModal } from '../components/modals';
+
+export const deleteHelmRelease = (releaseName: string, namespace: string, redirect?: string) => {
+  return {
+    label: 'Delete Helm Release',
+    callback: () => {
+      deleteResourceModal({
+        blocking: true,
+        resourceName: releaseName,
+        resourceType: 'Helm Release',
+        onSubmit: () => {
+          return coFetchJSON.delete(`/api/helm/release?name=${releaseName}&ns=${namespace}`);
+        },
+        close: redirect ? () => history.push(redirect) : undefined,
```

Comment:
Redirect occurs even if the operation is cancelled. This should only redirect on successful submission.

Reply:
Fixed it.